transcom/mymove
# [MB-16385] Adds authz check for moving expense PATCH endpoint

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16385)

## Summary

This pull request adds an authorization check to the UpdateMovingExpense service object, verifying that if a user using the internal API endpoint that calls the service object, they are the service member who is attached to the (orders -> move -> shipment -> PPM Shipment ->) moving expense that is being updated.

Various tests are added and edited for this functionality.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Verify that all server tests pass
2. Login to the Customer application as a user with a PPM shipment, and add a moving expense to that shipment. Verify that the moving expense is added correctly.
3. Start the process of adding another moving expense to that shipment (so that the URL changes to include the movingExpenseId fragment).
4. Using Postman and the credentials for another service member user, make a PATCH request to `/ppm-shipments/:ppmShipmentId/moving-expenses/:movingExpenseId` with the moving expense details in the body, and verify that a 404 is returned.

### Backend
